spec/controllers/activity_forms_controller_spec.rb

```diff
@@ -305,6 +305,26 @@
         end
       end
     end
+
+    describe "commitment" do
+      let(:user) { create(:beis_user) }
+
+      context "when it has not yet been set" do
+        let(:activity) { create(:programme_activity, commitment: nil) }
+
+        it "can be set" do
+          expect(get_step(:commitment)).to render_current_step
+        end
+      end
+
+      context "when it has already been set" do
+        let(:activity) { create(:programme_activity, commitment: create(:commitment, value: 1000)) }
+
+        it "cannot be edited" do
+          expect(get_step(:commitment)).to skip_to_next_step
+        end
+      end
+    end
```

Comment:
I'm not sure what you mean by expecting the opposite of planned behaviour? We're asserting we can view the form step to set the commitment. The `it` phrasing is slightly off here, as it's more about rendering the page rather than physically being able to set the commitment.

We prevent the page rendering [here](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/9b2f79654a091c9a4b2e68317f7ba93a33962f50/app/controllers/activity_forms_controller.rb#L81)

Reply:
Ah, makes more sense with the edit! I was confused because I was reading the `it`s as though it's allowing it to be set and preventing it to be edited from the activity details tab, but that's not the context. The update makes it clearer